atc/exec/artifact_output_step.go

```diff
@@ -44,24 +44,29 @@ func (step *ArtifactOutputStep) Run(ctx context.Context, state RunState) error {
 
 	outputName := step.plan.ArtifactOutput.Name
 
-	source, found := state.Artifacts().SourceFor(artifact.Name(outputName))
+	buildArtifact, found := state.ArtifactRepository().ArtifactFor(build.ArtifactName(outputName))
 	if !found {
 		return ArtifactNotFoundError{outputName}
 	}
 
-	volume, ok := source.(worker.Volume)
-	if !ok {
+	// TODO-Later Step shouldn't know about volumes
+	volume, found, err := step.workerClient.FindVolume(logger, step.build.TeamID(), buildArtifact.ID())
+	if err != nil {
+		return err
+	}
+
+	if !found {
 		return ArtifactNotFoundError{outputName}
 	}
 
-	artifact, err := volume.InitializeArtifact(outputName, step.build.ID())
+	dbWorkerArtifact, err := volume.InitializeArtifact(outputName, step.build.ID())
 	if err != nil {
 		return err
 	}
 
 	logger.Info("initialize-artifact-from-source", lager.Data{
```

Comment:
Is this log output still accurate? Looks like the term `source` was purged from elsewhere in this file